### ISSUE TITLE
How should I invoke examples with multiple HTTP routes?

### ISSUE THREAD

**Issue by timClicks** [awslabs/aws-lambda-rust-runtime#610]
The `http-axum` example provides a few HTTP accessible paths, but I'm unsure how to reach them via `cargo lambda invoke`. What data should I be providing?

**Comment by bnusunny** [comment]
Yes, you can use `curl`. It is in the README now. https://github.com/awslabs/aws-lambda-rust-runtime#cargo-lambda 

```
curl -v -X POST \
  'http://127.0.0.1:9001/lambda-url/<lambda-function-name>' \
  -H 'content-type: application/json' \
  -d '{ "command": "hi" }'
```

**Comment by timClicks** [comment]
Thanks for your answer @bnusunny. 

Two follow-on questions. How is `<lambda-function-name>` defined? I assume that it's the name of the `bin` target.

Is it possible to elaborate on what data should I be providing to the functions. How does the JSON payload map to something that the function handler can interpret?

**Comment by greenwoodcm** [comment]
if you use the `*/lambda-url/<lambda-function-name>/*` style then you can just curl as you would expect.  `cargo lambda` is going to do the translation of your curl request into the JSON payload, and then the Rust Runtime is going to do the de-translation of that JSON payload back into an HTTP request.

```
curl -X GET -H "Content-Type: application/json" http://localhost:9001/lambda-url/http-warp/foo/bar
```

**Comment by winstxnhdw** [comment]
Hey Tim! I was in the exact predicament as you a few weeks back.

> Two follow-on questions. How is <lambda-function-name> defined? I assume that it's the name of the bin target.

Yes, `lambda-function-name` is the name of the package defined in `Cargo.toml`. You can get the specifics from the [cargo-lambda invoke](https://www.cargo-lambda.info/commands/invoke.html) docs.

> Is it possible to elaborate on what data should I be providing to the functions. How does the JSON payload map to something that the function handler can interpret?

If you are using the `payload()` method, you can view the implementation [here](https://docs.rs/lambda_http/0.5.0/src/lambda_http/ext.rs.html#249). To elaborate further, if you expect the body of your request event to be of the following.

```json
{
    "words": ["hello", "world"],
    "number": 5
}
```

You should define a struct like so.

```rs
struct RequestEvent {
    words: Vec<String>,
    number: i32
}

async fn handler(event: Request) -> Result<Response<Body>, Error> {
    let payload = match event.payload::<RequestEvent>() {
        Ok(Some(payload)) => payload,
        Ok(None) => {
            return Ok(Response::builder()
                .status(400)
                .body("No body found".into())
                .map_err(Box::new)?)
        }
        Err(err) => {
            return Ok(Response::builder()
                .status(400)
                .body(format!("Error parsing body: {}", err).into())
                .map_err(Box::new)?)
        }
    };
}
```

Ideally, the entire docs should be completely rewritten and `cargo-lambda` should never be mentioned anywhere else apart from its dedicated section, but this is what we have to work with for now :)
